### ISSUE TITLE
[explicit-function-return-type] add an exclude list option

### ISSUE THREAD

**Issue by dangrussell** [typescript-eslint/typescript-eslint#4402]
### Description

Similar advantages to `allowedNames` on `explicit-module-boundary-types` and `ignoredMethodNames` on `explicit-member-accessibility`

### Fail

```ts
{
  "@typescript-eslint/explicit-function-return-type": [
    "error",
    {
      "allowedNames": []
    }
  ]
}

ngOnInit() {
    this.initStuff();
}

initStuff(): void {
   this.init = true;
}
```

### Pass

```ts
{
  "@typescript-eslint/explicit-function-return-type": [
    "error",
    {
      "allowedNames": ["ngOnInit"]
    }
  ]
}

ngOnInit() {
    this.initStuff();
}

initStuff(): void {
   this.init = true;
}
```

**Comment by hadimbj** [comment]
Or there are cases like Angular [lazy-loading](https://angular.io/guide/lazy-loading-ngmodules), It doesn't really makes sense to have a return type here. there should a way to exclude these.

```
const routes: Routes = [
  {
    path: 'items',
    loadChildren: () => import('./items/items.module').then(m => m.ItemsModule)
  }
];
```
